plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -319,15 +321,15 @@ public CompletableFuture<Collection<Slice>> finish()
     }
 
     // In spite of the name "Delta" Lake, we must rewrite the entire file to delete rows.
-    private List<Slice> rewriteFile(Path sourcePath, FileDeletion deletion)
+    private List<Slice> rewriteFile(String sourcePath, FileDeletion deletion)
     {
         try {
-            Path rootTablePath = new Path(rootTableLocation);
-            String sourceRelativePath = rootTablePath.toUri().relativize(sourcePath.toUri()).toString();
-
-            Path targetPath = new Path(sourcePath.getParent(), session.getQueryId() + "_" + randomUUID());
-            String targetRelativePath = rootTablePath.toUri().relativize(targetPath.toUri()).toString();
-            FileWriter fileWriter = createParquetFileWriter(targetPath.toString(), dataColumns);
+            URI rootTableUri = Locations.toUri(rootTableLocation);
```

Comment:
I assume that `rootTableLocation` is expected to be a prefix of `sourcePath`. We can verify that with `startsWith()` and then compute the target relative path using substring.